src/platform/packages/private/kbn-grid-layout/grid/use_grid_layout_state.ts

```diff
@@ -85,35 +97,42 @@ export const useGridLayoutState = ({
   }, [gridSettings, runtimeSettings$]);
 
   const gridLayoutStateManager = useMemo(() => {
-    const resolvedLayout = cloneDeep(layout);
-    Object.values(resolvedLayout).forEach((row) => {
-      resolvedLayout[row.id] = resolveGridRow(row);
-    });
-
-    const gridLayout$ = new BehaviorSubject<GridLayoutData>(resolvedLayout);
-    const proposedGridLayout$ = new BehaviorSubject<GridLayoutData | undefined>(undefined);
+    const orderedLayout = getOrderedLayout(layout);
+    const gridLayout$ = new BehaviorSubject<OrderedLayout>(orderedLayout);
     const gridDimensions$ = new BehaviorSubject<ObservedSize>({ width: 0, height: 0 });
-    const interactionEvent$ = new BehaviorSubject<PanelInteractionEvent | undefined>(undefined);
-    const activePanel$ = new BehaviorSubject<ActivePanel | undefined>(undefined);
-    const activeRowEvent$ = new BehaviorSubject<ActiveRowEvent | undefined>(undefined);
+    const activePanelEvent$ = new BehaviorSubject<ActivePanelEvent | undefined>(undefined);
+    const activeSectionEvent$ = new BehaviorSubject<ActiveSectionEvent | undefined>(undefined);
+
+    const layoutUpdated$: Observable<GridLayoutData> = combineLatest([
+      gridLayout$,
+      merge(activePanelEvent$, activeSectionEvent$),
+    ]).pipe(
+      // if an interaction event is happening, then ignore any "draft" layout changes
+      filter(([_, event]) => !Boolean(event)),
+      // once no interaction event, convert to the grid data format
+      map(([newLayout]) => getGridLayout(newLayout)),
+      // only emit if the layout has changed
+      distinctUntilChanged(isLayoutEqual)
+    );
 
     return {
       layoutRef,
-      rowRefs,
+      sectionRefs,
       headerRefs,
+      footerRefs,
```

Comment:
I think you ended up not using these so they can be removed

Reply:
Oops! You're totally right 🤦 Fixed in 4945ca7ca11d28edd7c77d834523c045099a912c